### PR TITLE
MM-60590: Add custom tags

### PR DESCRIPTION
**Pull request by agarciamontoro**
#### Summary
In order to better manage AWS resources, we need the ability to add custom tags. This will come in particularly handy for automated load-tests, where we want to tag each test in a unique way (using tags like `PR` or `SHA`), apart from tagging all automated load-tests in a consistent way (using a tag like `Origin` with a value like `automated-load-tests`).
This PR unblocks all this, adding a new `CustomTags` setting to the deployer that can receive an arbitrary amount of key-value pairs that are used to tag *all* resources deployed.

For example, using 
```json
"CustomTags": {
  "Origin": "agm-testing",
  "PR": "123",
  "SHA": "abcdefg"
}
```
we get all resources tagged as specified, apart from the already hard-coded `ClusterName` and `Name`. See how the first agent is tagged:
image

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-60590

### CODE REVIEW

File: config/deployer.sample.toml [hunk 1]
```diff
@@ -138,3 +138,5 @@ InstanceEngine = 'aurora-postgresql'
 InstanceType = 'db.r6g.large'
 Password = 'mostest80098bigpass_'
 UserName = 'mmuser'
+
+[CustomTags]
```
Comment (streamer45) [inline config/deployer.sample.toml:142]: Unrelated but I have to ask; is there anyone using this format? Why was it introduced?

File: deployment/terraform/assets/cluster.tf [hunk 2]
```diff
@@ -11,9 +11,12 @@ provider "aws" {
   region  = var.aws_region
   profile = var.aws_profile
   default_tags {
-    tags = {
-      ClusterName = var.cluster_name
-    }
+    tags = merge(
```
Comment (streamer45) [inline deployment/terraform/assets/cluster.tf:14]: nice

**Review by streamer45 — APPROVED** [review]
Makes sense 👍

File: config/deployer.sample.toml (hunk 1 above)
Comment (agnivade) [reply config/deployer.sample.toml:142]: Felipe added it. It's just for convenience I think more than anything.

Comment (agarciamontoro) [reply config/deployer.sample.toml:142]: Yup, this was a proposal from @fmartingr, which we gladly accepted as a potential UX improvement. I am not using it at the moment, but I want to test it and feel the difference :) Not sure if Felipe is using it currently.

File: deployment/config.go [hunk 3]
```diff
@@ -115,6 +115,21 @@ type Config struct {
...
 	// EnableNetPeekMetrics enables fine grained networking metrics collection through netpeek utility.
 	EnableNetPeekMetrics bool `default:"false"`
+	// CustomTags is an optional list of key-value pairs, which will be used as default
+	// tags for all resources deployed
+	CustomTags TerraformMap
+}
+
+// TerraformMap is a map of string -> string that serializes to the format expected by
+// the Terraform AWS provider when formatted as a string
+type TerraformMap map[string]string
```
Comment (agnivade) [inline deployment/config.go:125]: 0/5, but this map has nothing to do with Terraform right? I'd rather just say `StringMap`.

**Review by agnivade — APPROVED** [review]

File: deployment/config.go (hunk 3 above)
Comment (agarciamontoro) [reply deployment/config.go:125]: Its String method is Terraform-specific, that's why I called it that

**Pull request merged**